bundle/Command/ScheduledVisibilityUpdateCommand.php

```diff
@@ -0,0 +1,247 @@
+                continue;
+            }
+
+            try {
+                $contentId = $result['id'];
+
+                /** @var \Ibexa\Contracts\Core\Repository\Values\Content\Content $content */
+                $content = $this->repository->sudo(
+                    fn () => $this->repository->getContentService()->loadContent(
+                        $contentId,
+                        [$language->getLanguageCode()],
+                    ),
+                );
+            } catch (NotFoundException $exception) {
+                $this->logger->error(
+                    sprintf(
+                        'Content with id #%d does not exist: %s',
+                        $contentId,
+                        $exception->getMessage(),
+                    ),
+                );
+
+                continue;
+            }
+
+            try {
+                $action = $this->scheduledVisibilityService->updateVisibilityIfNeeded($content);
+            } catch (InvalidStateException $exception) {
+                $this->logger->error($exception->getMessage());
+
+                continue;
+            }
+
+            $this->style->progressAdvance();
+        }
+    }
+
+    private function getQueryBuilder(): QueryBuilder
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->select('id', 'initial_language_id')
+            ->from('ezcontentobject')
+            ->where('published != :unpublished')
+            ->orderBy('id', 'ASC')
+            ->setParameter('unpublished', 0);
+
+        return $query;
+    }
+
+    private function applyContentTypeLimit(QueryBuilder $query, array $contentTypeIds): void
+    {
+        $query->where(
+            $query->expr()->in('contentclass_id', ':content_type_ids'),
+        )->setParameter('content_type_ids', $contentTypeIds, Connection::PARAM_INT_ARRAY);
+    }
+
+    private function getPager(array $contentTypeIds): Pagerfanta
+    {
+        $query = $this->getQueryBuilder();
+
+        if (count($contentTypeIds) > 0) {
+            $this->applyContentTypeLimit($query, $contentTypeIds);
+        }
+
+        $countQueryBuilderModifier = function (QueryBuilder $queryBuilder) use ($contentTypeIds): void {
+            $queryBuilder->select('COUNT(id) AS total_results')
+                ->from('ezcontentobject')
+                ->where('published != :unpublished')
+                ->setParameter('unpublished', 0)
+                ->setMaxResults(1);
+
+            if (count($contentTypeIds) > 0) {
+                $this->applyContentTypeLimit($queryBuilder, $contentTypeIds);
+            }
+        };
+
+        return new Pagerfanta(new QueryAdapter($query, $countQueryBuilderModifier));
+    }
+
+    private function loadLanguage(int $id): Language
+    {
+        if (!isset($this->languageCache[$id])) {
+            $language = $this->repository->getContentLanguageService()->loadLanguageById($id);
+            $this->languageCache[$id] = $language;
```

Comment:
`$this->languageCache` is declared dynamically, it needs to be defined.

Reply:
fixed in d4e528b781023ddc9bb9a1201d0eb0fbfa0e3304